src/components/Project/index.js

```diff
@@ -118,6 +119,10 @@ export default class Project extends React.Component {
     if(loading){
       return (<Loading />)
     }
+    // handle invalid project
+    if(!project){
```

Comment:
Shouldn't be return this 404 page any time some route isn't handled, not only when the project doesn't exist?

Reply:
yup, that's what line 153 does

Reply:
I think I'm just missing the render flow of this. If we are rendering 404 on any unhandled route, why do we explicitly call it for the project?

Reply:
because project is used later on before render for setting other properties

Reply:
If we don't do this, we'll get an application error: "project undefined"

Reply:
Gotcha. Will this flash a view of the 404 if the project hasn't been pulled yet? Or does the `if loading(` clause take care of it?

There's probably a way of attaching route paths to resource ids, so when a resource ID is not found, the route automatically interprets it as a 404. Not sure how to implement it, but just something to watch out for.

Reply:
^ I agree, that's a good refactor solution. We could include as something to match in our route path and then show 404 as the default component if it isn't matched.